pyomo/contrib/mindtpy/util.py

```diff
@@ -9,6 +9,8 @@
 from pyomo.core.expr import current as EXPR
 from pyomo.opt import SolverFactory
 from pyomo.solvers.plugins.solvers.persistent_solver import PersistentSolver
+import cplex
```

Comment:
These imports should be made through _is_available. Currently, tests fail because CPLEX might not be available.

Reply:
I think we can refer to here.
https://github.com/Pyomo/pyomo/pull/1391#discussion_r415933190

Should I create a new file to place the incumbent callback?

Reply:
Yes, that should solve the problem.